frontend/public/widget.js

```diff
@@ -5,10 +5,14 @@
     const loadWidget = () => {
         window.hiEventWidgetLoaded = true;
 
-        let scriptOrigin;
+       let scriptOrigin;
         try {
-            const scriptURL = document.currentScript.src;
-            scriptOrigin = new URL(scriptURL).origin;
+let script = document.currentScript;
+if (!script) {
+   script = document.getElementById('hievents-widget-script');
+}```
+            if (!script || !script.src) throw new Error('HiEvent widget error: Invalid script URL');
+            scriptOrigin = new URL(script.src).origin;
```

Comment:
The formatting is incorrect here, and there are some quotes that break the code.